NYPLAudiobookToolkit/UI/AudiobookTableOfContentsTableViewController.swift

```diff
@@ -10,37 +10,94 @@ import UIKit
 
 let AudiobookTableOfContentsTableViewControllerCellIdentifier = "AudiobookTableOfContentsTableViewControllerCellIdentifier"
 
+public protocol AudiobookTableOfContentsTableViewControllerDelegate {
+    func userSelectedSpineItem(item: SpineElement)
+}
+
 public class AudiobookTableOfContentsTableViewController: UITableViewController, AudiobookTableOfContentsDelegate {
 
+
+    //MARK: - AudiobookTableOfContentsDelegate
+
     func audiobookTableOfContentsDidRequestReload(_ audiobookTableOfContents: AudiobookTableOfContents) {
-        self.tableView.reloadData()
+        if let selectedIndexPath = self.tableView.indexPathForSelectedRow {
+            self.tableView.reloadData()
+            self.tableView.selectRow(at: selectedIndexPath, animated: false, scrollPosition: .none)
+        } else {
+            self.tableView.reloadData()
+        }
+    }
+
+    func audiobookTableOfContentsPendingStatusDidUpdate(inProgress: Bool) {
+        if inProgress {
+            self.activityIndicator.startAnimating()
+        } else {
+            self.activityIndicator.stopAnimating()
+        }
+    }
+
+    func audiobookTableOfContentsUserSelected(spineItem: SpineElement) {
+        self.delegate?.userSelectedSpineItem(item: spineItem)
     }
 
+    //MARK: -
```

Comment:
what are we `//MARK: -`ing?

Reply:
Visual separation in the methods list:
![screen shot 2018-09-26 at 11 00 51 am](https://user-images.githubusercontent.com/6179695/46088929-73a2fe00-c17b-11e8-9990-f1d75f2c7505.png)

Compared to an extension-heavy source file (I don't know the protocols unless I know the methods really well):
![screen shot 2018-09-26 at 11 01 54 am](https://user-images.githubusercontent.com/6179695/46089023-a3ea9c80-c17b-11e8-82d6-20a021ce53a0.png)

I admit that there's probably a change in my usage behavior that could improve this..